casadm/cas_lib.c

```diff
@@ -1825,9 +1825,10 @@ int check_if_mounted(int cache_id, int core_id)
 	}
 
 	while ((mstruct = getmntent(mtab)) != NULL) {
+		mnt_fsname_len = strnlen(mstruct->mnt_fsname, sizeof(dev_buf));
 		/* mstruct->mnt_fsname is /dev/... block device path, not a mountpoint */
 		if ((NULL != mstruct->mnt_fsname)
-		    && (strncmp(mstruct->mnt_fsname, dev_buf, dev_buf_len) == 0)) {
+		    && (strncmp(mstruct->mnt_fsname, dev_buf, max(dev_buf_len, mnt_fsname_len)) == 0)) {
```

Comment:
this defeats the purpose of using strncmp here (although there are sufficient checks to make it safe anyways). For the sake of purity I would suggest strncpy_s(..., mnt_fsname_len, ..., dev_bug_len)